src/app/CommandHandlerInterfaceRegistry.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/CommandHandlerInterfaceRegistry.h>
+
+using namespace chip::app;
+
+namespace {
+
+CommandHandlerInterface * gCommandHandlerList = nullptr;
```

Comment:
Just wondering, didn't YY work towards having the old model with IME (using GetInstance()) to move away from using a global for registered commands?

Reply:
It is a step backwards here. My reason for making the step backwards was that after that, using the data model interface is a much larger step forward: the existing CHI in the InteractionModelEngine still does not cover the attribute access interface and the data model split would.

I will ask YY about this as well.

Reply:
I am okay with this if the later goal is to have this list one per IMEngine. This should still work for us temporarily since we do not manage multiple test nodes yet. The previous work I did is mostly for CommandHandlers to find the correct IMEngine and reverse is not really needed at the moment but definitely something we want in the future as well.